src/bsdfs/dielectric.cpp

```diff
@@ -337,6 +337,13 @@ class SmoothDielectric final : public BSDF<Float, Spectrum> {
         } else {
             if (likely(has_reflection && has_transmission)) {
                 weight = 1.f;
+                /* For differentiable variants, lobe choice has to be detached to avoid bias.
+                    Sampling weights should be computed accordingly. */
+                if constexpr (dr::is_diff_v<Float>) {
+                    if (dr::grad_enabled(r_i)) {
+                        weight = dr::select(selected_r, r_i / dr::detach(r_i), t_i / dr::detach(t_i));
```

Comment:
Another way to write the same logic
```c++
weight = dr::replace_grad(1.f, dr::select(selected_r, r_i, t_i));
```
Which will create a variable with a primal value of `1.f` and the gradient value corresponding to the one of the `select` statement.

Reply:
I'm not sure this is correct. The gradient of the weight has to be the gradient of the fraction `r_i / dr::detach(r_i)`.

Reply:
Ok then I take it back 👍